### ISSUE TITLE
FN: explicit_into_iter_loop: String::from("...").lines().into_iter()

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rust-clippy#2887]
````rust
let s = String::from("Hello
this 
is
a test");

let x = s.lines().into_iter();

for l in x {
    println!("line: {}", l);
}
````
the ````into_iter()```` is redundant but the lint does not catch it apparently.

In fact it does not even catch
````
    let x = s.lines().into_iter().into_iter().into_iter();
````
:/
clippy 0.0.211

**Comment by phansch** [comment]
Should be fixed by #2975
